core/PruManager.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ * PruManager.cpp
+ *
+ *	Created on: July 3, 2021
+ *		Author: Dhruva Gole
+ */
+
+#include "../include/PruManager.h"
+#include "MiscUtilities.h"
+#include <iostream>
+
+#if ENABLE_PRU_RPROC == 1
+PruManagerRprocMmap::PruManagerRprocMmap(unsigned int pruNum, unsigned int v)
+{
+	/* constructor for initializing the necessary path variables
+	 * based on the value of pru_num to choose:
+	 * 0 for PRU1 core 0
+	 * 1 for PRU1 core 1
+	 * 2 for PRU2 core 0
+	 * 3 for PRU2 core 1
+	 */
+	pru_num = pruNum;
+	verbose = v;
+	unsigned int pruss = pru_num / 2 + 1;
+	unsigned int prucore = pru_num % 2;
+
+	basePath = "/dev/remoteproc/pruss" + std::to_string(pruss) + "-core" + std::to_string(prucore) + "/";
+	statePath = basePath + "state";
+	firmwarePath = basePath + "firmware";
+	firmware = "am57xx-pru" + std::to_string(pruss) + "_" + std::to_string(prucore) + "-fw";
+# ifdef firmwareBelaRProc
+	std::string firmwareBela = firmwareBelaRProc;	// Incoming string from Makefile
+# endif	// firmwareBelaRProc
+	firmwareCopyCommand = "sudo ln -s -f " + firmwareBela + " /lib/firmware/" + firmware; // Pass the name of custom .out file via Makefile using firmwareBelaRProc=<path to file>
+	// 0 : pru1-core 0 in AI -> 4b234000
+	// 1 : pru1-core 1 in AI -> 4b238000
+	// 2 : pru2-core 0 in AI -> 4b2b4000
+	// 3 : pru2-core 1 in AI -> 4b2b8000
+	//
+	// 1 : PRUSS address in AI for PRU 1
+	// 2 : PRUSS address in AI for PRU 2
+	//
+	// 0 : pru-core 0 in BBB -> 4a334000
+	// 1 : pru-core 1 in BBB -> 4a338000
+
```

Comment:
one further thought, probably a better way to write this code is to have:
```
// in class declaration
std::vector<uint32_t> prussAddresses;
const std::vector<uint32_t> prussOwnRamOffsets = {0x0, 0x2000};
const uint32_t prussSharedRamOffset = 0x10000;

// in constructor
#ifdef IS_AM572x
prussAddresses.push_back(0x4b200000);
prussAddresses.push_back(0x4b280000);
#else // IS_AM572x
...
#endif // IS_AM572x
...
// in getOwnMemory()
	return ownMemory.map(prussAddresses[pruss] + prussOwnRamOffsets[prucore], 0x2000);
// in getSharedMemory()
	return sharedMemory.map(prussAddresses[pruss] + prussSharedRamOffset, 0x8000);

```

Reply:
> const std::vector<uint32_t> prussOwnRamOffsets = {0x0, 0x2000};

You probably mean  `{0x0, 0x4000};`? Since it's `0x4b234000` and `0x4b238000` for PRU1_0, PRU1_1 which is `prussAddresses[0] // =0x4b200000` + `0x4000` and + `0x8000` respectively.

Reply:
done in 2f8a6fcca96acc644a00bc40a58db96dbec3bf9f